Realm/RLMMigration.mm

```diff
@@ -142,6 +145,9 @@ - (void)execute:(RLMMigrationBlock)block {
 
         // verify uniqueness for any new unique columns before committing
         [self verifyPrimaryKeyUniqueness];
+
+        // reset schema to saved schema since it has been altered
+        RLMRealmSetSchema(_realm, savedSchema, true);
```

Comment:
Also while it's not needed right now, I think putting this above `verifyPrimaryKeyUniqueness` will be less likely to result in funny bugs if we change how `verifyPrimaryKeyUniqueness` works.